lib/ui/painting/utils.h

```diff
@@ -2,15 +2,35 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-#include "flutter/common/threads.h"
+#include "lib/ftl/synchronization/mutex.h"
 #include "third_party/skia/include/core/SkRefCnt.h"
 
+#include <queue>
+
 namespace blink {
 
+// A queue that holds Skia objects that must be destructed on the IO thread.
+class SkiaUnrefQueue {
+ public:
+  static SkiaUnrefQueue& Get();
+
+  void Unref(SkRefCnt* object);
+
+ private:
+  SkiaUnrefQueue();
+  void Drain();
+
+  static SkiaUnrefQueue instance_;
```

Comment:
We avoid static initialization. Maybe add a `SkiaUnrefQuue::Set` here that initializes the object once during engine initialization and add an assertion in the `Get`.